modules/apps/content-dashboard/content-dashboard-web/src/test/java/com/liferay/content/dashboard/web/internal/servlet/taglib/util/ContentDashboardDropdownItemsProviderTest.java

```diff
@@ -250,6 +253,103 @@ public void testGetURLBackURLWithBackURLParameter() {
 				String.valueOf(viewDropdownItem.get("href")), "p_l_back_url"));
 	}
 
+	@Test
+	public void testGetViewMetrics() {
+		ContentDashboardItemAction contentDashboardItemAction =
+			new ContentDashboardItemAction(
+				"View Metrics", "viewMetrics", "validURL");
+
+		ContentDashboardItemActionTracker contentDashboardItemActionTracker =
+			(className, classPK, httpServletRequest, locale) ->
+				Collections.singletonList(contentDashboardItemAction);
+
+		MockLiferayPortletRenderRequest mockLiferayPortletRenderRequest =
+			new MockLiferayPortletRenderRequest();
+
+		MockLiferayPortletURL mockLiferayPortletURL =
+			new MockLiferayPortletURL();
+
+		mockLiferayPortletRenderRequest.setAttribute(
+			"null" + StringPool.DASH + WebKeys.CURRENT_PORTLET_URL,
+			mockLiferayPortletURL);
+
+		mockLiferayPortletRenderRequest.setAttribute(
+			WebKeys.LOCALE, LocaleUtil.US);
+
+		ContentDashboardDropdownItemsProvider
+			contentDashboardDropdownItemsProvider =
+				new ContentDashboardDropdownItemsProvider(
+					contentDashboardItemActionTracker, _http, _language,
+					mockLiferayPortletRenderRequest,
+					new MockLiferayPortletRenderResponse(), new PortalImpl());
+
+		ContentDashboardItem contentDashboardItem = Mockito.mock(
+			ContentDashboardItem.class);
+
+		List<DropdownItem> dropdownItems =
+			contentDashboardDropdownItemsProvider.getDropdownItems(
+				contentDashboardItem);
+
+		Stream<DropdownItem> stream = dropdownItems.stream();
+
+		DropdownItem viewMetricsDropdownItem = stream.filter(
+			dropdownItem -> Objects.equals(
+				String.valueOf(dropdownItem.get("label")),
+				contentDashboardItemAction.getLabel())
+		).findFirst(
+		).orElseThrow(
+			AssertionError::new
+		);
+
+		Map<String, Object> data =
+			(Map<String, Object>)viewMetricsDropdownItem.get("data");
+
+		Assert.assertEquals(
+			contentDashboardItemAction.getName(), data.get("action"));
+
+		Assert.assertEquals(
+			contentDashboardItemAction.getUrl(), data.get("fetchURL"));
+	}
+
+	@Test
+	public void testGetViewMetricsWithEmptyContentDashboardItemActionTracker() {
```

Comment:
@cgoncas I'm not sure if this test adds value, what do you think?

Reply:
Of course @darquesdev, that is the only way we can test this API :)